manifests/init.pp

```diff
@@ -484,7 +484,7 @@
   }
 
   keycloak_conn_validator { 'keycloak':
-    keycloak_server => 'localhost',
+    keycloak_server => $service_bind_address,
```

Comment:
This won't work when the default is used which is `0.0.0.0`.  I think this needs to be a condition like this:

```puppet
if $service_bind_address in ['localhost','127.0.0.1','0.0.0.0'] {
  $validator_keycloak_server = 'localhost'
} else {
  $validator_keycloak_server = $service_bind_address
}
```

Then change `keycloak_server` to pull value from `$validator_keycloak_server`

Reply:
Hello @treydock,

to be honest I had exactly this discussion with a work colleague. The address `0.0.0.0` is resolved to the loopback address (`127.0.0.1`), I couldn't find anything on this behaviour in a hurry.
To rule out possible edge cases, I would provide condition handling for `0.0.0.0` as you suggested.

Greetings
Harm